src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,140 @@
+    private final Solenoid flap = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Conveyor.SOLENOID);
+    ColorMatch match = new ColorMatch();
+    private DriverStation.Alliance lastSeenColor = DriverStation.Alliance.Invalid;
+    private boolean lastPassed = true;
+
+    private Conveyor() {
+        motor.setInverted(TalonFXInvertType.Clockwise);
+        match.addColorMatch(Color.kRed); // red
+        match.addColorMatch(Color.kBlue); // blue
+    }
+
+    /**
+     * lazy instantiation
+     *
+     * @return the subsystem instance
+     */
+    public static Conveyor getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Conveyor();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * @return the color sensor value as a DriverStation.Alliance enum
+     */
+    public DriverStation.Alliance getColor() {
+        Color color = colorSensorIntake.getColor();
+        System.out.println(color.red + " Red " + color.blue + " Blue " + color.green + " Green ");
+        ColorMatchResult result = match.matchClosestColor(color);
+        Color resultColor = result.color;
+
+        if (resultColor == Color.kRed) {
+            return DriverStation.Alliance.Red;
+        } else if (resultColor == Color.kBlue) {
+            return DriverStation.Alliance.Blue;
+        } else {
+            return DriverStation.Alliance.Invalid;
+        }
+    }
+
+    /**
+     * @returns the amount of balls inside the cargo
+     */
+    public int getCargoCount() {
+        return position.size();
+    }
+
+    /**
+     * Set motor power.
+     * @param power the power to the motor [-1, 1].
+     */
+    public void setPower(double power) {
+        motor.set(power);
+    }
+
+    /**
+     * @param first the cargo place for which to get the color, true - first, false - last.
+     * @returns the color of first/last cargos inside the conveyor.
+     */
+    public DriverStation.Alliance getColorPosition(boolean first) {
+        if (first) {
+            return DriverStation.Alliance.valueOf(position.getFirst());
+        }
+        return DriverStation.Alliance.valueOf(position.getLast());
+    }
+
+    /**
+     * open the "flap" - solenoid
+     */
+    public void openFlap() {
+        flap.set(false);
+    }
+
+    /**
+     * closes the "flap" - solenoid
+     */
+    public void closeFlap() {
+        flap.set(true);
+    }
+
+    /**
+     * toggles the "flap" - solenoid
+     */
+    public void toggle() {
+        flap.toggle();
+    }
+
+    /**
+     *
+     * @param alliance an DriverStation.Alliance enum
+     * @return the name of the enum
+     */
+    private String enumToString(DriverStation.Alliance alliance) {
+        return alliance.name();
+    }
+
+    /**
+     * removes the string representing the cargo from the list if the cargo is ejected and adds if the cargo is consumed
+     */
+    @Override
+    public void periodic() {
+        var colorIntake = getColor();

```

Comment:
You need also to handle the case we outtake through the intake

Reply:
This code does handle this case

Reply:
This code does handle this case